src/key.cpp

```diff
@@ -74,23 +73,28 @@ CPubKey CKey::GetPubKey() const {
     return result;
 }
 
+extern "C"
```

Comment:
I wonder if we could somehow move this testcase-only code to src/tests instead of here.
Nah OTOH it's just a few lines. Any approach I can think of would either leak secp256k1 specifics on the interface or involve creating a larger wrapper.